app/utils/service_client/middleware/logger.rb

```diff
@@ -0,0 +1,24 @@
+module ServiceClient
+  module Middleware
+    class Logger < MiddlewareBase
+      def initialize
+        @logger = SharetribeLogger.new(:service_client)
+      end
+
+      def enter(ctx)
+        @logger.info("Enter", :enter, ctx)
+        ctx
+      end
+
+      def leave(ctx)
+        @logger.info("Leave", :leave, ctx)
+        ctx
+      end
+
+      def error(ctx)
+        @logger.info("Error", :error, ctx)
+        ctx
+      end
+    end
+  end
```

Comment:
I think a more useful logger implementation would be something that understands the request and response maps and can actually log what's happening. How about placing it close to end of queue where response is already built so you can log what kind of request will be made to which endpoint. I would also appreciate logging the time it takes from request to response. This is easy to do by putting the timestamp in the ctx in enter and logging diff in leave/error